### DISCUSSION TITLE
Custom object destroy

### DISCUSSION THREAD

**Discussion by ghost** [FirstGearGames/FishNet#480] [Feature Request]
I am spawning object with Addressables, and I am experiencing issues when I despawn a player.

So, my idea is to add a parameter called IObjectDestroyer with CustomDestroy(NetworkObject); method, that is called instead of regular Object.Destroy(...).

Thank you!

**Comment by ghost** [comment]
Example implementation:
- Add `Custom` field to `DespawnType` enum
- Add one more optional parameter `IObjectDestroyer` to `ServerManager.Despawn(NetworkObject, DespawnType? = null, IObjectDestroyer = null);`
- In the implementation, if passed `Custom` despawn type, ensure passed IObjectDestroyer != null, and when object is meant to be destroyed, use `IObjectDestroyer.CustomDestroy(NetworkObject);`

Or maybe name `IObjectDestroyer` to `INetworkObjectDestroyer` for some naming matching?

**Comment by FirstGearGames** [comment]
Bump

**Comment by FirstGearGames** [comment]
Revisiting this:
Is there any reason you cannot use OnNetworkStop? Despawning an object doesn't necessarily mean it will be destroyed, but calling Despawn will invoke OnNetworkStop.

**Comment by FirstGearGames** [comment]
Closing due to no reply.
